File: 数组/数组-简单/1431. 拥有最多糖果的孩子/1431. 拥有最多糖果的孩子.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */

 var kidsWithCandies = function(candies, extraCandies){
	var max = Math.max(...candies);
	return candies.map(candie => candie+extraCandies>=max)
};

File: 数组/数组-简单/1431. 拥有最多糖果的孩子/1431. 拥有最多糖果的孩子.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    max := 0
    for _, c := range candies {
        if c > max {
            max = c
        }
    }

    result := []bool{}
    for _, c := range candies {
        res := (c + extraCandies >= max)
        result = append(result, res)
    }

    return result
}
